apps/web/components/Charts/v2/hooks/data.ts

```diff
@@ -0,0 +1,62 @@
+import pick from 'lodash/pick'
+import { useQuery } from '@apollo/client'
+
+import {
+  Chart,
+  GetMultipleStatisticsQuery,
+  GetMultipleStatisticsQueryVariables,
+} from '@island.is/web/graphql/schema'
+import { GET_MULTIPLE_STATISTICS } from '@island.is/web/screens/queries/Statistics'
+
+import { ChartType } from '../types'
+
+type QueryResponse = {
+  getStatisticsByKeys: GetMultipleStatisticsQuery['getStatisticsByKeys']
+}
+
+interface UseGetChartDataProps {
+  dateFrom?: Chart['dateFrom']
+  dateTo?: Chart['dateTo']
+  numberOfDataPoints?: Chart['numberOfDataPoints']
+  components: {
+    sourceDataKey: string
+  }[]
+  chartType: ChartType | null
+}
+
+export const useGetChartData = ({
+  dateFrom,
+  dateTo,
+  numberOfDataPoints,
+  components,
+  chartType,
+}: UseGetChartDataProps) => {
+  const queryResult = useQuery<QueryResponse>(GET_MULTIPLE_STATISTICS, {
```

Comment:
Have you considered using graphql field resolvers instead of making a separate query on the client to get the graph data?

Reply:
I tried to make `Chart` a flexible drop-in embed for any rich text, and that's why they fetch their own data like this. If we switch to a field resolver, do we need to query the data separately for each page that uses them?

Reply:
If you look at the resolvers in the `libs/cms/src/lib/cms.resolver.ts` file you'll see that we can create a chart resolver so that everytime a chart is being resolved we can fetch it's relevant data. And that can be cached so if the same chart is being resolved within the cache time then the answer will just be there in the cache instead of us having to fetch it every time. However you are achieving this caching also when calling via the client but the user experience is a bit different.  

1. If we have a field resolver then the initial page will be more "complete" since the data will be there (but initial load time might increase if we get a cache miss)
2. If we fetch via the client then the user gets the initial page quicker but the charts will have to wait to render until there is an answer

I just wanted to mention that option 1 is available if you deem that better but honestly I think that option 2 is fine as well